### PR TITLE
add DDLStatement interface, extract command factory, tidy up, test etc

### PR DESCRIPTION
**Pull request by dguy**
Clean up code in `StatementExecutor`, `QueryEngine` etc.

### CODE REVIEW

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java [hunk 1]
```diff
@@ -286,6 +232,46 @@ private void executeStatement(
...
+                                      final Map<Long, CommandId> terminatedQueries,
+                                      final String statementStr) throws Exception {
+    QuerySpecification querySpecification =
+        (QuerySpecification) statement.getQuery().getQueryBody();
+    Query query = ksqlEngine.addInto(
+        statement.getQuery(),
+        querySpecification,
+        statement.getName().getSuffix(),
+        statement.getProperties(),
+        Optional.empty()
```
Comment (hjafarpour) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java]: I think this should be `statement.getPartitionByColumn()`.

Comment (dguy) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java]: Yep - thanks!

**Comment by dguy** [comment]
Thanks @hjafarpour, updated.

**Comment by hjafarpour** [comment]
LGTM.

**Pull request merged**